src/e2e/java/teammates/e2e/pageobjects/AdminNotificationsPage.java

```diff
@@ -0,0 +1,232 @@
+        // This is because the page will display all notifications in the database, which is not predictable
+        for (NotificationAttributes notification : notifications) {
+            WebElement notificationRow = notificationsTable.findElement(By.id(notification.getNotificationId()));
+            verifyTableRowValues(notificationRow, getNotificationTableDisplayDetails(notification));
+        }
+    }
+
+    public void verifyNotificationAttributes(NotificationAttributes expected, NotificationAttributes actual) {
+        // Note: Notification ID is not checked
+        // as the actual notification has a randomly generated ID
+        assertEquals(expected.getStartTime(), actual.getStartTime());
+        assertEquals(expected.getEndTime(), actual.getEndTime());
+        assertEquals(expected.getStyle(), actual.getStyle());
+        assertEquals(expected.getTargetUser(), actual.getTargetUser());
+        assertEquals(expected.getTitle(), actual.getTitle());
+        assertEquals(expected.getMessage(), actual.getMessage());
+    }
+
+    public void addNotification(NotificationAttributes notification) {
+        clickAddNotificationButton();
+        waitForElementPresence(By.id("btn-create-notification"));
+
+        fillNotificationForm(notification);
+
+        clickCreateNotificationButton();
+        waitForPageToLoad(true);
+    }
+
+    public void editNotification(NotificationAttributes notification) {
+        WebElement notificationRow = notificationsTable.findElement(By.id(notification.getNotificationId()));
+        WebElement editButton = notificationRow.findElement(By.className("btn-light"));
+        editButton.click();
+        waitForElementPresence(By.id("btn-edit-notification"));
+
+        fillNotificationForm(notification);
+
+        clickEditNotificationButton();
+        waitForPageToLoad(true);
+    }
+
+    public void deleteNotification(NotificationAttributes notification) {
+        WebElement notificationRow = notificationsTable.findElement(By.id(notification.getNotificationId()));
+        WebElement deleteButton = notificationRow.findElement(By.className("btn-danger"));
+
+        deleteButton.click();
+        waitForConfirmationModalAndClickOk();
+        waitForPageToLoad(true);
+    }
+
+    public void fillNotificationForm(NotificationAttributes notification) {
+        selectDropdownOptionByText(notificationTargetUserDropdown, getTargetUserText(notification.getTargetUser()));
+        selectDropdownOptionByText(notificationStyleDropdown, getNotificationStyle(notification.getStyle()));
+        fillTextBox(notificationTitleTextBox, notification.getTitle());
+        setMessage(notification.getMessage());
+        setNotificationStartDateTime(notification.getStartTime());
+        setNotificationEndDateTime(notification.getEndTime());
+    }
+
+    public String getNewestNotificationId() {
```

Comment:
Would this necessarily be the newest notification? Seems like it would be possible to have a few elements with "table-success".  Should we change it to something like `getANewNotificationId()` perhaps add a comment as well?

Reply:
`findElement` returns the first element encountered that has the given attribute, and since the newest created notification is at the top row, the notification returned would be the newest notification even if there are multiple rows of class `table-success`. I'll add a comment for this!

Reply:
My bad this does make sense, I was testing it with edit function. I think the comment would still be good since if we refresh the page and edit an older comment, it would not necessarily take the top comment

Reply:
Good suggestion! I've modified the logic such that it sorts the table by creation time before taking the ID of the first row, this way it should guarantee that the latest notification ID is returned